gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -54,6 +78,87 @@ public class GoogleCloudStorageClientImpl extends ForwardingGoogleCloudStorage {
             .setHttpRequestInitializer(httpRequestInitializer)
             .setDownscopedAccessTokenFn(downscopedAccessTokenFn)
             .build());
+    this.storageOptions = checkNotNull(options, "options must not be null");
+    this.storage =
+        clientLibraryStorage == null ? createStorage(credentials, options) : clientLibraryStorage;
+  }
+
+  @Override
+  public WritableByteChannel create(StorageResourceId resourceId, CreateObjectOptions options)
+      throws IOException {
+    if (storageOptions.isGrpcEnabled()) {
```

Comment:
We can but we wouldn't till we are confident and have done a performance testing on new client library. Hence falling back to old implementation.

Reply:
Hm, but in this case why customer will even enable new client if they can use old implementation w/o it? I think that for simplicity we always need to use new gRPC code path for new client, otherwise it's confusing how these configs interact.

Reply:
Hmm, We anyhow use delegates wherever we don't have existing implementation for new client.
But faire point have another if to decide on which is getting used adds a bit of confusion.

Will always follow gRPC code path for new client if implementation is overriden.